docs/docs/migration/v0.25.0.md

```diff
@@ -64,4 +64,934 @@ ignite s module mars
 
 The only difference is the additional directory `planet` which is the name of the application.  The name of the app can 
 be verified by checking the package in the `go.mod` file.  In this example, the package is `github.com/cosmos/planet` where
-`planet` is the app name.
\ No newline at end of file
+`planet` is the app name.
+
+ ---
+
+## Removing `cosmoscmd`
+
+`v0.25.0` removes the `cosmoscmd` package from scaffolded chains.  This package provided  utility for creating 
+commands and starting up their application.  The `cosmoscmd` package is now deprecated, and it is suggested that chains
+import this functionality in their codebase so that their chains can be more easily upgraded and customized.
+
+The main functionality of `cosmoscmd` will be moved to the `app` package of your chain.  Some imports in these 
+examples contain the sample string, `{ModulePath}`.  Replace this string with the Go module path of your blockchain.
+For example, if your blockchain module path is `github.com/planet/mars`, `{ModulePath}/app/params` would be become
+`github.com/planet/mars/app/params`.
+
+#### Migration in `app` package
+
+To begin, create  a new file, `./app/params/encoding.go`, containing the following code:
+
+```go
+package params
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+// EncodingConfig specifies the concrete encoding types to use for a given app.
+// This is provided for compatibility between protobuf and amino implementations.
+type EncodingConfig struct {
+	InterfaceRegistry types.InterfaceRegistry
+	Marshaler         codec.Codec
+	TxConfig          client.TxConfig
+	Amino             *codec.LegacyAmino
+}
+```
+
+Next, create  a new file, `./app/encoding.go`, containing the following code:
+
+```go
+package app
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
+
+	"{ModulePath}/app/params"
+)
+
+// makeEncodingConfig creates an EncodingConfig for an amino based test configuration.
+func makeEncodingConfig() params.EncodingConfig {
```

Comment:
This function is deprecated in GAIAD 
https://github.com/cosmos/gaia/blob/main/app/encoding.go#L12